### ISSUE TITLE
Wrong return type in `load_balancing_loss_func` function of `modeling_mixtral.py`.

### ISSUE THREAD

**Issue by PhilipMay** [huggingface/transformers#33639]
Hi.

this return type here is wrong:

https://github.com/huggingface/transformers/blob/78b2929c0554b79e0489b451ce4ece14d265ead2/src/transformers/models/mixtral/modeling_mixtral.py#L127

It is not `float` but `torch.Tensor`.

I think this should be fixed as it is confusing and can lead to errors in use. 

What do you think?
PR is at #33641

**Comment by LysandreJik** [comment]
Thanks a lot for your PR @PhilipMay :raised_hands:

**Comment by af-74413592** [comment]
i think tokens_per_expert = torch.sum(expert_mask.float() * expert_attention_mask, dim=0) / torch.sum(
            expert_attention_mask, dim=0
        )    miss count the topk dim.

**Comment by claralp** [comment]
@af-74413592 @PhilipMay seems to be `dim=(0,1)` to me as well
you can check the result with a simple example and set e.g.:
```python
num_experts = 4
top_k = 2
gate_logits = tuple(torch.randn(2, 16, 4))
attention_mask = torch.ones((2, 16))
```
